### ISSUE TITLE
Frames are skipping

### ISSUE THREAD

**Issue by JuniorDjjr** [cleolibrary/CLEO-Redux#21]
https://youtu.be/_crRD00DPIM  

The movement should be smooth, after all, I'm using wait 0.

```
{$cleo}
0000:

while true
    wait 0      

    if 0C00: key_pressed 89 //y
    then
        04C4: store_coords_to 11@ 12@ 13@ from_actor $PLAYER_ACTOR with_offset 0.0 1.0 0.0
        00A1: put_actor $PLAYER_ACTOR at 11@ 12@ 13@
    end  
    if 0C00: key_pressed 72 //h
    then
        04C4: store_coords_to 11@ 12@ 13@ from_actor $PLAYER_ACTOR with_offset 0.0 -1.0 0.0
        00A1: put_actor $PLAYER_ACTOR at 11@ 12@ 13@
    end
    
end

**Comment by x87** [comment]
Does it happen when you run the same script in main.scm?

(note that 0C00 won't work in main.scm, but you can use widget keys for testing)

**Comment by JuniorDjjr** [comment]
I haven't tested it, but now I better understand the cause of the problem.

Try making 100 copies of this line:
`04C4: store_coords_to 11@ 12@ 13@ from_actor $PLAYER_ACTOR with_offset 0.0 1.0 0.0`
And hold Y.
The script will be slower to process, as a result, it takes around 3 seconds to process the loop and take effect.

It's like a lazy processing, I don't know. The longer the script, the longer it takes to process. Quickly takes a matter of seconds.

**Comment by x87** [comment]
this is expected, no? game runs scripts in main loop, which means while it executes opcodes it can't render a new frame. this is why we have wait command to put the script on hold and let the game finish rendering a frame. the more opcodes you add in between wait commands, the slower things are.

**Comment by JuniorDjjr** [comment]
But that's not how script execution works, just when "wait" is read the next game frame is executed.  
You can loop 10,000 commands, the game will never skip frames.

This is for PC:
```
{$cleo}
0000:

while true
    wait 0
    
    if 0AB0:   is_key_pressed 89 //y, PC opcode
    then
        0@ = 0
        while 0@ < 10000
            0@ += 1
            0099: 1@ = random_int_in_ranges_0_to_32767 // just to make it slower
        end
        04C4: store_coords_to 11@ 12@ 13@ from_actor $PLAYER_ACTOR with_offset 0.0 -10.0 0.0
        00A1: put_actor $PLAYER_ACTOR at 11@ 12@ 13@
    end
    
end

I lowered this script to only 1,000 iterations, tested it on SA:DE, pressed Y **and the game ran for over 1 minute until it took effect**. One possibility is that each command read, is executed 1 "wait 0" (roughly saying, to understand it). This script above would take more than 10 minutes to take effect in SA:DE.

**Comment by x87** [comment]
Can you test the same on pure main.scm?

**Comment by x87** [comment]
I can confirm the player movement is smooth on main.scm and jerking on cleo. I will look into what may cause it

**Comment by XMDS** [comment]
Remove wait 0 and it will become faster. I mentioned this problem before, but there is no time to study in these two days.

**Comment by x87** [comment]
@XMDS  Loops must have wait command. It's actually a CLEO bug and a critical one, I must say.

@JuniorDjjr thanks for providing a test script and hinting the main.scm behavior. I was able to identify and fix the problem. Check the new dev build on discord.

**Comment by JuniorDjjr** [comment]
It's working perfectly now, I was able to create a free camera mod without any problems.

Taking the advantage, a minor issue: 007B: in Sanny Builder for "SA Mobile" is written = instead of +=

**Comment by x87** [comment]
fixed in 0.8.3
